File: frontend/lib/config/departments.dart
// DEPRECATED: Use ConfigService instead
// These are fallback values only
// Always fetch from backend using ConfigService.getDepartments()

/// Fallback departments (use ConfigService for dynamic values)
const List<String> fallbackDepartments = [
  'Municipal Cleanliness',
  'Electrical Department',
  'Water Department',
  'Roads & Infrastructure',
  'Health & Sanitation',
];

/// Fallback priorities (use ConfigService for dynamic values)
const List<String> fallbackPriorities = [
  'low',
  'medium',
  'high',
  'urgent',
];

/// Fallback statuses (use ConfigService for dynamic values)
const List<String> fallbackStatuses = [
  'submitted',
  'assigned',
  'in_progress',
  'resolved',
  'closed',
  'rejected',
];

// For backward compatibility (deprecated - use ConfigService)
@Deprecated('Use ConfigService.getDepartments() instead')
const List<String> departments = fallbackDepartments;

@Deprecated('Use ConfigService.getPriorities() instead')
const List<String> priorities = fallbackPriorities;

@Deprecated('Use ConfigService.getStatuses() instead')
const List<String> statuses = fallbackStatuses;

File: backend/api/config/departments.js
/**
 * Campus-specific departments
 * Same infrastructure/maintenance departments as village app, but for campus context
 */

export const DEPARTMENTS = [
  'Municipal Cleanliness',      // Garbage, waste, cleaning issues on campus
  'Electrical Department',       // Power outages, electrical hazards, street lights
  'Water Department',            // Water supply, leaks, drainage, flooding
  'Roads & Infrastructure',      // Potholes, road damage, pathways, sidewalks
  'Health & Sanitation'          // Public health issues, sanitation problems, medical emergencies
];

export const PRIORITY_LEVELS = ['low', 'medium', 'high', 'urgent'];

export const STATUS_LEVELS = [
  'submitted',
  'assigned',
  'in_progress',
  'resolved',
  'closed',
  'rejected'
];

export const PRIORITY_CONFIG = {
  low: { label: 'Low', color: 'priority-low' },
  medium: { label: 'Medium', color: 'priority-medium' },
  high: { label: 'High', color: 'priority-high' },
  urgent: { label: 'Urgent', color: 'priority-urgent' }
};

export const STATUS_CONFIG = {
  submitted: { label: 'Submitted', color: 'status-submitted' },
  assigned: { label: 'Assigned', color: 'status-assigned' },
  in_progress: { label: 'In Progress', color: 'status-in-progress' },
  resolved: { label: 'Resolved', color: 'status-resolved' },
  closed: { label: 'Closed', color: 'status-closed' },
  rejected: { label: 'Rejected', color: 'status-rejected' }
};
